java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/JsonDeserializer.java

```diff
@@ -167,6 +193,40 @@ private static Object validateJsonDataTypes(Data.DataType.TypeName typeName, Obj
                             .asRuntimeException();
                 }
                 return Timestamp.valueOf((String) value);
+            case TIME:
+                return castTime(value);
+            case DATE:
+                return castDate(value);
+            case INTERVAL:
+                if (!(value instanceof String)) {
+                    throw io.grpc.Status.INVALID_ARGUMENT
+                            .withDescription("Expected interval, got " + value.getClass())
+                            .asRuntimeException();
+                }
+                return value;
+            case JSONB:
+                if (!(value instanceof String)) {
+                    throw io.grpc.Status.INVALID_ARGUMENT
+                            .withDescription("Expected jsonb, got " + value.getClass())
+                            .asRuntimeException();
+                }
+                return value;
+            case BYTEA:
+                if (!(value instanceof String)) {
+                    throw io.grpc.Status.INVALID_ARGUMENT
+                            .withDescription("Expected jsonb, got " + value.getClass())
+                            .asRuntimeException();
+                }
+                String s = (String) value;
+                int len = s.length();
+                byte[] bytes = new byte[len / 2];
+                for (int i = 0; i < len; i += 2) {
```

Comment:
We should use base64 encoding which is more compact. @WillyKidd

Reply:
I thought we have abandoned `JsonDeserializer` and moved to `StreamChunkDeserializer` but it seems not 🤣 When do you estimate we'll finish it?

Reply:
I discussed this issue with Yiming before. Currently, we use a global config `RW_CONNECTOR_RPC_SINK_PAYLOAD_FORMAT` to set the payload format, JSON  by default. IMO, we can set the format to stream_chunk for the `Remote` sink (sink via the connector node). But the challenge is that the compute node relies on a .so built from `src/java_binding`, which is not considered in the deployment right now. cc @wenym1

Reply:
In addition, before we switch to the `stream_chunk` format, we should conduct a performance evaluation to compare it with JSON. I will look into it later.